benchmarks/pipe.py

```diff
@@ -444,30 +467,30 @@ def benchmark_single_process(args):
     init_random_seed(0)
 
     benchmark_config = create_benchmark_config(args.model_name)
-    model_config = make_model_and_data(args, config=benchmark_config)
+    model_config = create_model_config(args, config=benchmark_config)
     model = model_config["model"]
 
     balance = generate_balance(min(num_devices, 4), len(model))
-    p = pipe.Pipe(
+    pipe_model = pipe.Pipe(
         model, balance, chunks=args.chunks, pipelined_backward=args.pipelined_backward, checkpoint=args.checkpoint
     )
     del model
     del model_config["model"]
 
     if args.use_synthetic_data:
-        train(model_config, p, benchmark_config, args)
+        train(model_config, pipe_model, benchmark_config, args)
     else:
-        benchmark_language_model(model_config, p, benchmark_config, args)
+        benchmark_language_model(model_config, pipe_model, benchmark_config, args)
```

Comment:
How about add a comment like:

`# args.use_synthetic_data implies that we don't need to benchmark. Hence, we call train() directly without the benchmark_language_model wrapper.`

Since the functions are called the same way, it is better to reduce duplicate a bit:

func = train
if not args.use_synthetic_data:
    func = benchmark_language_model
func(...)

Reply:
I want to have users to be able to benchmark synthetic data as well. The flow that I am thinking of is the following:
- introduce a dry_run flag where we don't compare values and instead just train the model
- use_synthetic_data will switch between using real vs synthetic data as the name suggests.

benchmark_single_process should look something like
model, optimizer = get_model_and_optimizer
dataloader = get_dataloader
benchmark_config = get_benchmark_config

benchmark_model(model, optimizer, dataloader, benchmark_config)

Reply:
sounds good!